tests/page-objects/default/enketo/pregnancy.wdio.page.js

```diff
@@ -0,0 +1,185 @@
+  `span[data-value=" /pregnancy/weeks_since_lmp_rounded "] `);
+const futureVisitDate = () => $(`${FORM} ` +
+  `section[name="/pregnancy/anc_visits_hf/anc_visits_hf_next/anc_next_visit_date"] input.ignore.input-small`);
+const riskFactors = (value) => $$(`${FORM} ` +
+  `input[name="/pregnancy/risk_factors/risk_factors_present/${value}_condition"]`);
+const patientNameSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/patient_name "]`);
+const weeksPregnantSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/weeks_since_lmp_rounded "]`);
+const eddSummary = () => $(`${SUMMARY_SECTION} span[data-value=" /pregnancy/summary/edd_summary "]`);
+const riskFactorsSummary = () => $$(`${SUMMARY_SECTION} ` +
+  `:not(label.disabled) span[data-itext-id*="/pregnancy/summary/r_risk"]`);
+const dangerSignsSummary = () => $$(`${SUMMARY_SECTION} ` +
+  `:not(label.disabled) span[data-itext-id*="/pregnancy/summary/r_danger_sign_"]`);
+
+const selectGestationAge = async (value = GEST_AGE.edd) => {
+  const getAge = await gestationalAge(value);
+  await getAge.waitForDisplayed();
+  await getAge.click();
+};
+
+const setDeliveryDate = async (value = moment().add(1, 'month').format('YYYY-MM-DD')) => {
+  const date = await deliveryDate();
+  await date.waitForDisplayed();
+  await date.setValue(value);
+};
+
+const getConfirmationDetails = async () => {
+  return {
+    eddConfirm: await eddConfirmation().getText(),
+    weeksPregnantConfirm: await weeksPregnantConfirmation().getText(),
+  };
+};
+
+const setANCVisitsPast = async (value = 0) => {
+  const visits = await ancVisitsPast();
+  await visits.waitForDisplayed();
+  await visits.setValue(value);
+};
+
+const setFutureVisitDate = async (value = moment().add(1, 'day').format('YYYY-MM-DD')) => {
+  const date = await futureVisitDate();
+  await date.waitForDisplayed();
+  await date.setValue(value);
+};
+
+const selectYesNoOption = async (selector, value = 'yes') => {
+  const element = await $(`${FORM} ${selector}[value="${value}"]`);
+  await element.waitForDisplayed();
+  await element.click();
+  return value === 'yes';
+};
+
+// The selector for the 'riskFactors' is different depending on the option selected in the 'first pregnancy' question
+const selectAllRiskFactors = async (value = FIRST_PREGNANCY_VALUE.no) => {
+  const cbRisks = await riskFactors(value);
+  for (const risk of cbRisks) {
+    if(await risk.getAttribute('value') !== 'none'){
+      await risk.click();
+    }
+  }
+  return cbRisks.length - 1; // Subtract 1 due to the 'none' option
+};
+
+const getSummaryDetails = async () => {
+  return {
+    patientNameSumm: await patientNameSummary().getText(),
+    weeksPregnantSumm: await weeksPregnantSummary().getText(),
+    eddSumm: await eddSummary().getText(),
+    riskFactorsSumm: await riskFactorsSummary().length - 1, // Subtract 1 due to the title
+    dangerSignsSumm: await dangerSignsSummary().length,
+  };
+};
+
+const submitPregnancy = async () => {
```

Comment:
If we need to make another test case with different data for the same form, what do we do?

Reply:
You can use the new version of the method 😅 
Thanks @latin-panda